src/components/FileUploadModal/upload-local-files.ts

```diff
@@ -15,104 +15,100 @@ import { VIEWER_ERROR } from "../../state/viewer/constants";
-A "custom request" for an Antd Upload component is a function that overrides
-the default POST request that happens when the user uploads files. The Antd
-Upload is a wrapper around the Upload component from react-component (rc-upload).
+/**
+Takes in an array of all the files loaded by the user, finds and processes the
+.simularium file and any geometry files separately, then tells the app to load
+the trajectory with its associated geometry files into the viewer.
+
+This function may be used as a "custom request" for an Antd Upload component,
+which overrides the default POST request that happens when the user uploads
+files. The Antd Upload wraps the Upload component from react-component.
 - Antd docs: https://ant.design/components/upload/#API
 - rc-upload docs: https://github.com/react-component/upload#customrequest
-
-This custom request function takes in an array of all the files loaded by the
-user, finds and processes the .simularium file and any geometry files separately, 
-then tells the app to load the trajectory with its associated geometry files 
-into the viewer.
 */
-export default (
-    { onSuccess, onError }: UploadRequestOption,
+export default async (
     selectedFiles: File[],
     clearSimulariumFile: (isNewFile: { newFile: boolean }) => void,
     loadFunction: (simulariumFile: LocalSimFile) => void,
     setViewerStatus: (status: { status: ViewerStatus }) => void,
-    setError: (error: ViewerError) => void
+    setError: (error: ViewerError) => void,
+    rcRequest?: UploadRequestOption
 ) => {
-    numCustomRequests++;
-    if (numCustomRequests !== 1) {
+    numRequests++;
+    if (numRequests !== 1) {
         // If the user loads multiple files at once (.simularium file + geometry file(s)),
         // this function is called multiple times, but we only need to process
         // and load the trajectory once.
-        if (numCustomRequests === selectedFiles.length) {
+        if (numRequests === selectedFiles.length) {
             // We're at the last customRequest for this batch of files,
             // so reset counter
-            numCustomRequests = 0;
+            numRequests = 0;
         }
         return;
     }
 
     if (selectedFiles.length === 1) {
         // numCustomRequests and selectedFiles.length are both 1, so reset
-        numCustomRequests = 0;
+        numRequests = 0;
     }
 
     // want the loading indicator to show without any lag time
     // as soon as user hits "Open" button or drops files,
     // and not have to have this action called multiple places in the code.
     clearSimulariumFile({ newFile: true });
 
-    const filesArr: File[] = Array.from(selectedFiles) as File[];
-
     try {
         // Try to identify the simularium file.
         // Put all the other files as text based geoAssets.
-        const simulariumFileIndex = findIndex(filesArr, (file) =>
+        const simulariumFileIndex = findIndex(selectedFiles, (file) =>
             file.name.includes(".simularium")
```

Comment:
Should this be `endsWith`? Are there instances where the file extension would be something other than .simularium, or where a file would contain .simularium in its name before the extension?

Reply:
yeah it could be endsWith